### DISCUSSION TITLE
Guest Auth

### DISCUSSION THREAD

**Discussion by argon338** [casdoor/casdoor#2053] [Q&A] [answered]
If I need to make guest auth? The only nickname and fingerprint from fronted as query and to send back JWT to user. 
Of course to make a user profile linked by uuid in JWT. 
Any ideas how to add custom auth to casdoor?

**Comment by ISulimanov** [comment]
I guess it something like a "anonymous session" like[ firebase](https://firebase.google.com/docs/auth/web/anonymous-auth)

As far as I know [Zitadel](https://github.com/zitadel/zitadel/discussions/5304) or even keycloak does not support it so far. 

If Casdoor will able to provide easy way to do it - it will be good competitive advantage.

**Reply by argon338** [reply]
In my case it is similar to zoom guest participant for conference. User came to it, set name, some kind of preferences and maybe it was browser crash or user reloaded page and we need to keep information about that user. Or maybe later user will upgrade his account with mobile, email for OAuth. 

But I was absolutely sure casdoor has some mechanics to link golang files as an additional scenarios

**Comment by hsluoyz** [comment, marked as answer]
@argon338 @ISulimanov done at: https://github.com/casdoor/casdoor/pull/4421
